core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -129,16 +129,17 @@ class BrokerMetadataListener(
     }
   }
 
-  private def shouldSnapshot(): Option[String] = {
+  private def shouldSnapshot(): Option[SnapshotReason] = {
     val metadataVersionHasChanged: Boolean = metadataVersionChanged()
     val maxBytesHaveExceeded: Boolean = (_bytesSinceLastSnapshot >= maxBytesBetweenSnapshots)
 
+    //TODO: Figure out how to handle multiple reasons for snapshot
```

Comment:
One solution is to return a `Set[SnapshotReason]` and pass the set when creating the snapshot.